File: 2015/solutions/day1.py
with open('../inputs/input1.txt', 'r') as f:
    data = f.read()

    # part one
    print(data.count('(') - data.count(')'))

    # part two
    pos = 1
    floor = 0
    for c in data:
        if c == '(':
            floor += 1
        else:
            floor -= 1
        if floor == -1:
            print(pos)
            break
        pos += 1

File: 2017/solutions/day1.js
const fs = require('fs');

let summation = (a, b) => {return parseInt(a) + parseInt(b)};

fs.readFile('../inputs/input1.txt', 'utf-8', (err, data) => {
    data = data.trim();
    let re = /(.)(?=\1)/g;
    let sum = data.match(re).reduce(summation);
    if (data.charAt(0) === data.slice(-1)) sum += parseInt(data.charAt(0));

    //part one
    console.log(sum);

    let half = data.length / 2
    re = new RegExp(`(.)(?=.{${half - 1}}\\1)`, 'g');
    let looped = data.slice(half) + data.slice(0, half);
    sum = data.match(re).reduce(summation);
    sum += looped.match(re).reduce(summation);

    //part two
    console.log(sum);
});